util/nocasemaps/nocase.go

```diff
@@ -74,6 +74,23 @@ func Delete[K ~string, V any](m map[K]V, k K) {
 	delete(m, K(appendToLower(a[:0], string(k))))
 }
 
+// Append is equivalent to:
+//
+//	append(m[strings.ToLower(k)], v)
+func Append[K ~string, S []E, E any](m map[K]S, k K, vs ...E) {
```

Comment:
Bikeshed: I think `AppendElem` or `AppendSlice` or `AppendSliceElem` is a clearer name as `Append` is not an operation that makes sense for maps.